### PR TITLE
feat: support basic authorization

### LINKED ISSUE #497: Feat: Provide methods for logging into the karpor dashboard
### What would you like to be added?

Limit the login of Karpor Dashboard, only the dashboard can be viewed only by adopting the authentication. You can considers the **Service Account Token** or **OIDC**.

### Why is this needed?

Restrict the user group that can view cluster information.

### PR DESCRIPTION
**Pull request by ruquanzhao**
## What type of PR is this?
/kind feature

## What this PR does / why we need it:

This PR provides the login functionality, which is essential in a production environment.

## Which issue(s) this PR fixes:

Fixes #497

### CODE REVIEW

File: pkg/core/route/route.go [hunk 1]
```diff
@@ -105,9 +106,12 @@ func NewCoreRoute(
 	// Endpoint to list all available endpoints in the router.
 	router.Get("/endpoints", endpointhandler.Endpoints(router))
 
-	// Endpoint to list all available endpoints in the router.
+	// Expose
```
Comment (elliotxx) [inline pkg/core/route/route.go:109]: Comments should be restored

File: cmd/app/options/recommended.go [hunk 2]
```diff
@@ -16,11 +16,13 @@ package options
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/spf13/pflag"
 
 	karporopenapi "github.com/KusionStack/karpor/pkg/kubernetes/generated/openapi"
```
Comment (elliotxx) [inline cmd/app/options/recommended.go:23]: need to format

Comment (ruquanzhao) [reply cmd/app/options/recommended.go:21]: Done

**Comment by elliotxx** [comment]
lgtm

**Review by elliotxx — APPROVED** [review]

**Pull request merged**